src/data/scripts/world/systems/CMC_PeachGarden.java

```diff
@@ -74,10 +77,14 @@ public void generate(SectorAPI sector) {
                 true);
         //make a custom description which is specified in descriptions.csv
         yuanming.setCustomDescriptionId("cmc_planet_yuanming");
+        Industry yuanmingOrbitalWorks = yuanmingMarket.getIndustry(Industries.ORBITALWORKS);
         //give the orbital works a gamma core
-        yuanmingMarket.getIndustry(Industries.ORBITALWORKS).setAICoreId(Commodities.GAMMA_CORE);
-        //and give it a nanoforge
-        ((HeavyIndustry) yuanmingMarket.getIndustry(Industries.ORBITALWORKS)).setNanoforge(new SpecialItemData(Items.CORRUPTED_NANOFORGE, null));
+        yuanmingOrbitalWorks.setAICoreId(Commodities.GAMMA_CORE);
+        // give it a nanoforge
+        yuanmingOrbitalWorks.setSpecialItem(new SpecialItemData(Items.CORRUPTED_NANOFORGE, null));
+        // and apply its effects
+        InstallableItemEffect itemEffect = ItemEffectsRepo.ITEM_EFFECTS.get(Items.CORRUPTED_NANOFORGE);
```

Comment:
You could now have corrupted nanoforge effect without having corrupted nanoforge installed. Would be "unstealable" by Nex.